Packages/com.unity.inputsystem/InputSystem/Actions/InputActionAsset.cs

```diff
@@ -924,8 +924,8 @@ private void OnDestroy()
         [NonSerialized] internal InputActionState m_SharedStateForAllMaps;
         [NonSerialized] internal InputBinding? m_BindingMask;
 
-        [NonSerialized] private int m_DevicesCount = -1;
-        [NonSerialized] private InputDevice[] m_DevicesArray;
+        [NonSerialized] internal int m_DevicesCount = -1;
+        [NonSerialized] internal InputDevice[] m_DevicesArray;
```

Comment:
Not sure. IMO sharing internals between a very small set of closely related classes is not an issue. The coupled is straightforwardly visible from the references and refactoring in the future, should the need arise, is straightforward. Downside to encapsulating the data is you pay for the encapsulation with more methods and more code.

But yeah, would definitely agree there's strict limits to this. Once it's not just one or two tightly coupled classes but rather something seeping out all over the place, it's getting stupid.

Reply:
It's only shared between two closely related classes **now**, but by making it internal you're saying to the next person that writes code relating to it that it's ok to use it directly, which it's really not. Also encapsulating it will result in less code, not more. And if we're at the point of worrying about the cost of a method call, it's time to pack up and write everything in assembly :) So basically, in my only slightly dogmatic book of coding, this is the wrong thing to do, and I would only approve it if you're adamant you don't want to change it.

Reply:
>if we're at the point of worrying about the cost of a method call

Nah, not the cost of the call. Meant just the code.

I'll go wrap it.

Reply:
Ok, finally fixed this. Indeed much better now. And also found some bugs in the original code in the process.